control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go

```diff
@@ -7,51 +7,70 @@ import (
 
 const (
 	CloudConfigDir      = "/etc/openstack/config"
-	CloudCredentialsDir = "/etc/openstack"
-	CredentialsFile     = "clouds.conf"
+	CloudCredentialsDir = "/etc/openstack/secret"
+	CredentialsFile     = "cloud.conf"
 	CaDir               = "/etc/pki/ca-trust/extracted/pem"
-	CaKey               = "ca.pem"
+	CABundleKey         = "ca-bundle.pem"
 	Provider            = "openstack"
-	cloudsSecretKey     = "clouds.yaml"
-	caSecretKey         = "cacert"
+	CloudsSecretKey     = "clouds.yaml"
+	CASecretKey         = "cacert"
 )
 
-// ReconcileCloudConfig reconciles as expected by Nodes Kubelet.
-func ReconcileCloudConfig(secret *corev1.Secret, hcp *hyperv1.HostedControlPlane, credentialsSecret *corev1.Secret, hasCACert bool) error {
+// ReconcileCloudConfig reconciles the cloud config secret.
+func ReconcileCloudConfigSecret(secret *corev1.Secret, cloudName string, credentialsSecret *corev1.Secret, caCertData []byte) error {
 	if secret.Data == nil {
 		secret.Data = map[string][]byte{}
 	}
-	config := string(credentialsSecret.Data[CredentialsFile]) // TODO(emilien): Missing key handling
-
-	config += `
-[Global]
-use-clouds=true
-clouds-file=` + CloudCredentialsDir + "/" + cloudsSecretKey + "\n"
+	config := getCloudConfig(cloudName, credentialsSecret, caCertData)
+	if caCertData != nil {
+		secret.Data[CABundleKey] = caCertData
+	}
+	secret.Data[CredentialsFile] = []byte(config)
 
-	config += "\ncloud=" + hcp.Spec.Platform.OpenStack.IdentityRef.CloudName
+	return nil
+}
 
-	if hasCACert {
-		config += "\nca-file =" + CaDir + "/" + CaKey + "\n"
+// ReconcileCloudConfigConfigMap reconciles the cloud config configmap.
+func ReconcileCloudConfigConfigMap(cm *corev1.ConfigMap, cloudName string, credentialsSecret *corev1.Secret, caCertData []byte) error {
```

Comment:
This and the `ReconcileCloudConfigSecret` contain pretty much the same thing. Do we need to have two separate resources to hold the same data?

Reply:
one is for generating a secret, the other is a configmap. I'm adding comments to explain better. Thanks for pointing that out.